cashu/wallet/helpers.py

```diff
@@ -0,0 +1,453 @@
+
+        # ----- receive token -----
+
+        # deserialize token
+        # dtoken = json.loads(base64.urlsafe_b64decode(token))
+        tokenObj = TokenV3.deserialize(token)
+
+        # tokenObj = TokenV2.parse_obj(dtoken)
+        if is_api:
+            assert len(tokenObj.token), HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST, detail="no proofs in token."
+            )
+            assert len(tokenObj.token[0].proofs), HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST, detail="no proofs in token"
+            )
+        else:
+            assert len(tokenObj.token), Exception("no proofs in token")
+            assert len(tokenObj.token[0].proofs), Exception("no proofs in token")
+        includes_mint_info: bool = any([t.mint for t in tokenObj.token])
+
+        # if there is a `mints` field in the token
+        # we check whether the token has mints that we don't know yet
+        # and ask the user if they want to trust the new mints
+        if includes_mint_info:
+            # we ask the user to confirm any new mints the tokens may include
+            # await verify_mints(ctx, tokenObj)
+            # redeem tokens with new wallet instances
+            await redeem_TokenV3_multimint(
+                wallet, tokenObj, script, signature, is_api=is_api
+            )
+        else:
+            # no mint information present, we extract the proofs and use wallet's default mint
+
+            proofs = [p for t in tokenObj.token for p in t.proofs]
+            # first we load the mint URL from the DB
+            keyset_in_token = proofs[0].id
+            assert keyset_in_token
+            # we get the keyset from the db
+            mint_keysets = await get_keyset(id=keyset_in_token, db=wallet.db)
+            if is_api:
+                assert mint_keysets, HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="we don't know this keyset",
+                )
+                assert mint_keysets.mint_url, HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="we don't know this mint's URL",
+                )
+            else:
+                assert mint_keysets, Exception("we don't know this keyset")
+                assert mint_keysets.mint_url, Exception("we don't know this mint's URL")
+            # now we have the URL
+            mint_wallet = Wallet(
+                mint_keysets.mint_url,
+                os.path.join(settings.cashu_dir, wallet.name),
+            )
+            await mint_wallet.load_mint(keyset_in_token)
+            _, _ = await mint_wallet.redeem(proofs, script, signature)
+            if not is_api:
+                print(f"Received {sum_proofs(proofs)} sats")
+
+    # reload main wallet so the balance updates
+    await wallet.load_proofs()
+    if is_api:
+        return wallet.available_balance
+    else:
+        wallet.status()
```

Comment:
Function should not return two different types.

Reply:
What would be better is to leave functions in `cli_helpers` or `helpers.py` as unmodified as possible (except maybe for the contexts) but handle all changes to business logic (in this case you want `wallet.status()`) outside of these files, somewhere in `api/`.